pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/CountAggregationFunction.java

```diff
@@ -89,7 +122,33 @@ public void aggregate(int length, AggregationResultHolder aggregationResultHolde
   @Override
   public void aggregateGroupBySV(int length, int[] groupKeyArray, GroupByResultHolder groupByResultHolder,
       Map<ExpressionContext, BlockValSet> blockValSetMap) {
-    if (blockValSetMap.isEmpty()) {
+    if (blockValSetMap.isEmpty() || !blockValSetMap.containsKey(STAR_TREE_COUNT_STAR_EXPRESSION)) {
```

Comment:
@Jackie-Jiang We have three branches:
1- blockValSetMap.isEmpty() for count(*), and no StarTree.
2- !blockValSetMap.containsKey(STAR_TREE_COUNT_STAR_EXPRESSION) for count(columnName), and no StarTree (please see the ! at the beginning). Example: count(bigDecimalColumn). Note: we can have count(bigDecimalColumn) and nullHandling is either enable or not enabled.
3- StarTree.

Reply:
When both null handling and star-tree are not enabled, will it pass in non-empty `blockValSetMap`? Ideally it should pass in empty `blockValSetMap`, or it could add extra overhead for existing use cases

Reply:
I think I find the reason. When null handling is not enabled, ideally `getInputExpressions()` should return empty list to preserve the existing behavior, which is changed because we changed it to extend `BaseSingleInputAggregationFunction`. Since most methods in `BaseSingleInputAggregationFunction` cannot be reused, suggest not extending it, and keep the existing behavior when null handling is not enabled

Reply:
Applied a commit to demonstrate the idea: https://github.com/apache/pinot/pull/9086/commits/b6845345692bf0e1f3bb9bdd8f80d9eec606da8a
Let's see if it can pass all the tests

Reply:
The overhead is should be minimal. I don't have strong opinion either way. Will apply your commit.